### PR TITLE
Improved COSU experience

### PR DESCRIPTION
**Pull request by janakamarasena**

### CODE REVIEW

**Review by chaturadilan — APPROVED** [review]

File: client/client/src/main/java/org/wso2/iot/agent/AuthenticationActivity.java [hunk 1]
```diff
@@ -225,6 +227,22 @@ public void onClick(DialogInterface dialog, int whichButton) {
 			});
 		}
 
+		footer = (LinearLayout) findViewById(R.id.footer);
+		if (Constants.COSU_SECRET_EXIT) {
+			footer.setOnClickListener(new View.OnClickListener() {
+				@Override
+				public void onClick(View v) {
+					kioskExit++;
+					if (kioskExit == 6) {
```
Comment (ruwanta) [inline client/client/src/main/java/org/wso2/iot/agent/AuthenticationActivity.java:236]: Magic number, need to have class level constant with proper name.

File: client/client/src/main/java/org/wso2/iot/agent/KioskActivity.java [hunk 2]
```diff
@@ -132,6 +137,19 @@ private void startPolling() {
         }
     }
 
+    /*Checks whether there is an already installed app and if exists the app will be launched*/
```
Comment (ruwanta) [inline client/client/src/main/java/org/wso2/iot/agent/KioskActivity.java:140]: Comment is not in proper style

File: client/client/src/main/java/org/wso2/iot/agent/KioskActivity.java [hunk 3]
```diff
@@ -132,6 +137,19 @@ private void startPolling() {
...
+    private void launchKioskAppIfExists() {
+        packageName = Preference.getString(context.getApplicationContext(), kioskAppPackageNameKey);
+        if (packageName != null && !packageName.equals("")) {
```
Comment (ruwanta) [inline client/client/src/main/java/org/wso2/iot/agent/KioskActivity.java:143]: can use StringUtils.isEmpty()

File: client/client/src/main/java/org/wso2/iot/agent/services/AgentStartupReceiver.java [hunk 4]
```diff
@@ -69,6 +70,13 @@ public void onReceive(final Context context, Intent intent) {
 			Intent serviceIntent = new Intent(context, LocationService.class);
 			context.startService(serviceIntent);
 		}
+		if (intent.getAction().equals(Intent.ACTION_BOOT_COMPLETED)) {
+			if(Constants.DEFAULT_OWNERSHIP == Constants.OWNERSHIP_COSU) {
```
Comment (ruwanta) [inline client/client/src/main/java/org/wso2/iot/agent/services/AgentStartupReceiver.java:74]: if(Constants.DEFAULT_OWNERSHIP == Constants.OWNERSHIP_COSU)
looks this is always evaluate to single value

**Review by ruwanta — COMMENTED** [review]
Some comments.

File: client/client/src/main/java/org/wso2/iot/agent/services/AgentStartupReceiver.java (hunk 4 above)
Comment (janakamarasena) [reply client/client/src/main/java/org/wso2/iot/agent/services/AgentStartupReceiver.java:74]: This code block has been changed.

File: client/client/src/main/java/org/wso2/iot/agent/KioskActivity.java (hunk 3 above)
Comment (janakamarasena) [reply client/client/src/main/java/org/wso2/iot/agent/KioskActivity.java:143]: Fixed with https://github.com/wso2/cdmf-agent-android/pull/199

File: client/client/src/main/java/org/wso2/iot/agent/KioskActivity.java (hunk 2 above)
Comment (janakamarasena) [reply client/client/src/main/java/org/wso2/iot/agent/KioskActivity.java:140]: Fixed with https://github.com/wso2/cdmf-agent-android/pull/199

File: client/client/src/main/java/org/wso2/iot/agent/AuthenticationActivity.java (hunk 1 above)
Comment (janakamarasena) [reply client/client/src/main/java/org/wso2/iot/agent/AuthenticationActivity.java:236]: Fixed with https://github.com/wso2/cdmf-agent-android/pull/199

File: client/client/src/main/java/org/wso2/iot/agent/KioskActivity.java [hunk 5]
```diff
@@ -132,6 +137,19 @@ private void startPolling() {
         }
     }
 
+    /*Checks whether there is an already installed app and if exists the app will be launched*/
+    private void launchKioskAppIfExists() {
```
Comment (nilasini) [inline client/client/src/main/java/org/wso2/iot/agent/KioskActivity.java:141]: Add new line after method signature

Comment (janakamarasena) [reply client/client/src/main/java/org/wso2/iot/agent/KioskActivity.java:141]: Fixed with https://github.com/wso2/cdmf-agent-android/pull/199

File: client/client/src/main/java/org/wso2/iot/agent/KioskActivity.java [hunk 6]
```diff
@@ -40,11 +41,13 @@
     int kioskExit;
 
     static String packageName = null;
+    private String kioskAppPackageNameKey ="kioskAppPackageName";
```
Comment (sinthuja) [inline client/client/src/main/java/org/wso2/iot/agent/KioskActivity.java:44]: Formatting issue, there should be a space after the =

Comment (janakamarasena) [reply client/client/src/main/java/org/wso2/iot/agent/KioskActivity.java:44]: This code is no longer available in the latest file.

**Pull request merged**